rockets/jsonrpc/requestProcessor.cpp

```diff
@@ -60,18 +62,28 @@ inline std::string dump(const json& object)
 void RequestProcessor::process(const Request& request,
                                AsyncStringResponse callback)
 {
-    const auto document = json::parse(request.message, nullptr, false);
-    if (document.is_object())
+    try
     {
-        auto stringifyCallback = [callback](const json obj) {
-            callback(dump(obj));
-        };
-        _processCommand(document, request.clientID, stringifyCallback);
+        const auto document = json::parse(request.message);
+        if (document.is_object())
+        {
+            auto stringifyCallback = [callback](const json obj) {
+                callback(dump(obj));
+            };
+            _processCommand(document, request.clientID, stringifyCallback);
+        }
+        else if (document.is_array())
+            callback(_processBatchBlocking(document, request.clientID));
+        else
+        {
+            const auto id = document.count("id") ? document["id"] : json();
```

Comment:
can there really ever be an id given that document is not an object..?

Reply:
True, does not make sense actually.